bundle/config/mutator/python/merge_output.go

```diff
@@ -0,0 +1,194 @@
+package python
+
+import (
+	"fmt"
+
+	"github.com/databricks/cli/bundle/config/mutator"
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/databricks/cli/libs/dyn/merge"
+)
+
+type mergeResult struct {
+	AddedResources   mutator.ResourceKeySet
+	UpdatedResources mutator.ResourceKeySet
+	DeletedResources mutator.ResourceKeySet
+}
+
+// mergeOutput merges output of Python mutator with the input configuration.
+//
+// It records which resources have been added and updated into visitorState.
+func mergeOutput(root, output dyn.Value) (dyn.Value, mergeResult, error) {
```

Comment:
Would be good to clarify that
1. root -- existing state
2. output -- output from python.
3. this function returns output but locations are kept from the input
4. It also fixes empty collections behavior, if empty collection is missing in the output, it is restored.